### PR TITLE
Add enabled toggle

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add enabled toggle

## why
* Only if enabled=true will this module create resources

## references
N/A

### CODE REVIEW

**Review by osterman — APPROVED** [review]

**Review by max-lobur — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -15,9 +19,9 @@ resource "aws_sns_topic" "this" {
 }
 
 resource "aws_sns_topic_subscription" "this" {
-  for_each = var.subscribers
+  for_each = local.enabled ? var.subscribers : {}
```
Comment (aknysh) [inline main.tf:22]: @nitrocode  did you test it with `enabled=false`?
usually in this case terraform complains that both sides of the `?` operator must have the same types (not only it must be a map, but all items must be the same with the same types)

**Review by aknysh — COMMENTED** [review]
please see comments

File: main.tf (hunk 1 above)
Comment (aknysh) [reply main.tf:22]: > var.subscribers is a map so both types are the same

yes, but in similar cases Terraform complains that not only the container types must be the same (maps), but they must have the the items and the same item types

Comment (nitrocode) [reply main.tf:22]: What would you propose to use here instead ?

**Review by aknysh — APPROVED** [review]

**Pull request merged**